src/main/java/com/streamr/client/authentication/ApiKeyAuthenticationMethod.java

```diff
@@ -20,10 +22,21 @@ public ApiKeyAuthenticationMethod(String apiKey) {
     @Override
     protected LoginResponse login(String restApiUrl) throws IOException {
         Response response = null;
+        ResponseBody body = null;
+        BufferedSource source = null;
         try {
             response = post(restApiUrl + "/login/apikey", adapter.toJson(new ApiKeyLoginRequest(apiKey)));
-            return parse(response.body().source());
+            body = response.body();
+            source = body.source();
+            LoginResponse loginResponse = parse(source);
+            return loginResponse;
         } finally {
+            if (source != null) {
```

Comment:
We can remove the duplicated request parsing+closing logic if we call `StreamrRESTClient.execute()` here. Same applied to `login()` and` challenge()` methods in `EthereumAuthenticationMethod`.

Reply:
https://linear.app/streamr/issue/NET-145/duplicated-http-request-parsing-logic-in-java-client